loop/utils/generators.py

```
@@ -0,0 +1,10 @@
+import numpy as np
+
+def generate_permutation(params):
+    
+    out_dict = {}
+
+    for key in params().keys():
```

Ruff (0.8.2)

Use `key in dict` instead of `key in dict.keys()`

Remove `.keys()`

(SIM118)